pkg/helm/watcher/controller/helm_watcher.go

```diff
@@ -88,9 +88,11 @@ func (r *HelmWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 				continue
 			}
 
-			values[chart.Name] = map[string][]byte{
-				v: valueBytes,
+			if _, ok := values[chart.Name]; !ok {
+				values[chart.Name] = make(map[string][]byte)
 			}
+
+			values[chart.Name][v] = valueBytes
```

Comment:
Yeah, this was a quick pr so SImon is unlocked, I'm writing more tests. It's a bit meh, because I have to return a different value on a different call :) So I'm setting up the mock. Gimme a second :D

Reply:
:ok_hand:

Reply:
Thanks for the thought! I'm not super blocked by it right now, I can use the latests version for now.

Reply:
@aclevername oh this was easier than I thought. :D Can you take a look now? Does it make sense?

So for each version there is a different value per the same profile.